poly/src/polynomial/multivariate/sparse.rs

```diff
@@ -0,0 +1,403 @@
+//! A sparse multivariate polynomial represented in coefficient form.
+use crate::{
+    multivariate::{SparseTerm, Term},
+    MVPolynomial, Polynomial,
+};
+use ark_ff::Field;
+use ark_std::{
+    cmp::Ordering,
+    fmt,
+    ops::{Add, AddAssign, Neg, Sub, SubAssign},
+    vec::Vec,
+};
+use rand::Rng;
+
+#[cfg(feature = "parallel")]
+use rayon::prelude::*;
+
+/// Stores a sparse multivariate polynomial in coefficient form.
+#[derive(Derivative)]
+#[derivative(Clone, PartialEq, Eq, Hash, Default)]
+pub struct SparsePolynomial<F: Field, T: Term> {
+    /// The number of variables the polynomial supports
+    #[derivative(PartialEq = "ignore")]
+    pub num_vars: usize,
+    /// List of every term along with its coefficient
+    pub terms: Vec<(T, F)>,
```

Comment:
I think storing in `(coeff, term)` form would be clearer as it maps to how you write the polynomial, eg: p(x, y) = 2 * (x^2 * y^2)`.

```suggestion
    pub terms: Vec<(F, T)>,
```